anta/reporter/md_reporter.py

```diff
@@ -0,0 +1,332 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Markdown report generator for ANTA test results."""
+
+from __future__ import annotations
+
+import logging
+import re
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, ClassVar
+
+from anta.constants import MD_REPORT_TOC
+from anta.logger import anta_log_exception
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+    from io import TextIOWrapper
+    from pathlib import Path
+
+    from anta.result_manager import ResultManager
+
+logger = logging.getLogger(__name__)
+
+
+# pylint: disable=too-few-public-methods
+class MDReportGenerator:
+    """Class responsible for generating a Markdown report based on the provided `ResultManager` object.
+
+    It aggregates different report sections, each represented by a subclass of `MDReportBase`,
+    and sequentially generates their content into a markdown file.
+
+    The `generate` class method will loop over all the section subclasses and call their `generate_section` method.
+    The final report will be generated in the same order as the `sections` list of the method.
+
+    The class method also accepts an optional `only_failed_tests` flag to generate a report with only failed tests.
+
+    By default, the report will include all test results.
```

Comment:
lets use the `--hide` content instead for greater versatility

Reply:
Fixed in 8918cd0822dee86db2c1574d30732c0418bfefff